app/javascript/listings/listings.jsx

```diff
@@ -284,27 +301,28 @@ export class Listings extends Component {
 
   listingSearch(query, tags, category, slug) {
     const t = this;
-    const { index, page, listings } = t.state;
-    const filterObject = { tagFilters: tags, hitsPerPage: 75, page };
-    if (category.length > 0) {
-      filterObject.filters = `category:${category}`;
-    }
-    index.search(query, filterObject).then(function searchDone(content) {
-      const fullListings = listings;
-      content.hits.forEach(listing => {
-        if (listing.bumped_at) {
-          if (!listings.map(l => l.id).includes(listing.id)) {
-            fullListings.push(listing);
-          }
-        }
-      });
+    const { page, listings } = t.state;
+    const dataHash = {
+      per_page: 75,
+      page,
+      classified_listing_search: query,
+    };
+
+    if (dataHash.tags) dataHash.tags = tags;
+
+    if (category.length > 0) dataHash.category = category;
+
+    const responsePromise = fetchSearch('classified_listings', dataHash);
+    return responsePromise.then(response => {
+      const classifiedListings = response.result;
+      const fullListings = updateListings(classifiedListings, listings);
       t.setState({
         listings: fullListings,
         initialFetch: false,
-        showNextPageButt: content.hits.length === 75,
+        showNextPageButt: classifiedListings.length === 75,
```

Comment:
What is `75` here? This feels like a magic number to me. Can we make it a `const` at the top of the file with some documentation for it, and then use the const here?

Reply:
Update!